contracts/src/v0.8/functions/dev/v1_X/FunctionsBilling.sol

```diff
@@ -261,14 +261,6 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
   ) internal returns (FunctionsResponse.FulfillResult) {
     FunctionsResponse.Commitment memory commitment = abi.decode(onchainMetadata, (FunctionsResponse.Commitment));
 
-    if (s_requestCommitments[requestId] == bytes32(0)) {
```

Comment:
Can you explain in the PR description that these are repeated inside fulfill() and triggering them over there is better because it also emits events with error messages?

Also, why are tests such as "FunctionsRouter_Fulfill:test_Fulfill_RequestNotProcessedInvalidRequestId()" not affected gas-wise? They should be using slightly more now, no?

Reply:
Added.

The FunctionsRouter test doesn't go through the FunctionsCoordinator, it calls the Router directly, so it is unaffected by Coordinator changes

Reply:
I see. How ward would it be to add a simple test that goes through coordinator into router.fulfill()? I just want to be extra safe with coverage.

Reply:
I'll get a PR up that will add ^ before we merge this one, then we can see it change in this PR

Reply:
Actually you can see it changing on the Gas Minimum Request & Gas Maximum Request tests, those go through the end to end flow - we should probably just merge this, as I am adding more to the test PR that I mentioned